### ISSUE TITLE
LuceneDataSource fails to release write lock when recovering from an unclean database shutdown in Neo4j embedded 2.2.2 and above

### ISSUE THREAD

**Issue by cmorgner** [neo4j/neo4j#4957]
Hi,

we're using Neo4j 2.2.3 embedded with several Lucene indexes that are created on startup. It seems that when the database is started again after an unclean shutdown (kill -9, system crash, etc.), the LuceneCommandApplier acquires a write lock on the data source in visitIndexDefineCommand(), but never releases the lock (it seems that the close() method is never called, verified by debugging).

The result of this issue is that the Lucene index is completely locked, no query can be executed until the system is shut down cleanly, and started again.

I can confirm that the behavior is NOT present in 2.1.8, but in 2.2.2 and 2.2.3. I have not been able to create test code to reproduce the bug, because it seems to depend on more factors than just an unclean shutdown. It does not occur when using a minimal test setup with an embedded GraphDatabaseService and an index.

Here is how we create our indexes, are we doing something wrong here?

```
final Index<Node> fulltextIndex = graphDb.index().forNodes("fulltextIndex", LuceneIndexImplementation.FULLTEXT_CONFIG);
final Index<Node> keywordIndex  = graphDb.index().forNodes("keywordIndex", LuceneIndexImplementation.EXACT_CONFIG);
```

We do not synchronize access to the index objects. Can that be a problem?

The messages.log does not contain any unusual messages.

It's not a blocker for us, but maybe there is a quick solution as this seems like a small bug.

Thank you and best regards,
Christian Morgner

/CC @amorgner

**Comment by cmorgner** [comment]
I just realized that it's LuceneDataSource instead of LuceneDataStore, sorry for that..

**Comment by cmorgner** [comment]
The issue is already fixed in commit c9596af0d990d7dde9a9053799b7295985ae3c05 from 10 days ago, thanks a lot! :)
